src/pages/RFOX/components/Shared/TransactionRow.tsx

```diff
@@ -0,0 +1,153 @@
+import {
+  Button,
+  Card,
+  CardBody,
+  CardHeader,
+  Center,
+  Collapse,
+  Flex,
+  Link,
+  Text,
+} from '@chakra-ui/react'
+import type { AssetId } from '@shapeshiftoss/caip'
+import { fromAssetId } from '@shapeshiftoss/caip'
+import { SwapperName } from '@shapeshiftoss/swapper'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import { useCallback, useMemo, useState } from 'react'
+import { FaCheck } from 'react-icons/fa'
+import { FaX } from 'react-icons/fa6'
+import { useTranslate } from 'react-polyglot'
+import { AssetIcon } from 'components/AssetIcon'
+import { CircularProgress } from 'components/CircularProgress/CircularProgress'
+import { getTxLink } from 'lib/getTxLink'
+import { selectAssetById, selectFeeAssetByChainId, selectTxById } from 'state/slices/selectors'
+import { useAppSelector } from 'state/store'
+
+import type { MultiStepStatusStep } from './SharedMultiStepStatus'
+
+type TransactionRowProps = {
+  assetId: AssetId
+  onStart: () => void
+  headerCopy: string
+  isActive?: boolean
+  txId?: string
+  serializedTxIndex: string | undefined
+} & Pick<MultiStepStatusStep, 'isActionable' | 'onSignAndBroadcast'>
+
+export const TransactionRow: React.FC<TransactionRowProps> = ({
+  assetId,
+  headerCopy,
+  onStart,
+  isActive,
+  isActionable,
+  serializedTxIndex,
+  onSignAndBroadcast,
+  txId,
+}) => {
+  const translate = useTranslate()
+
+  const tx = useAppSelector(state => selectTxById(state, serializedTxIndex ?? ''))
+  const status = useMemo(() => tx?.status, [tx?.status])
+  const [isSubmitting, setIsSubmitting] = useState(false)
+
+  const asset = useAppSelector(state => selectAssetById(state, assetId))
+  const feeAsset = useAppSelector(state =>
+    selectFeeAssetByChainId(state, fromAssetId(assetId).chainId),
+  )
+
+  const txIdLink = useMemo(
+    () =>
+      getTxLink({
+        defaultExplorerBaseUrl: asset?.explorerTxLink ?? '',
+        tradeId: txId ?? '',
+        name: SwapperName.ArbitrumBridge,
+      }),
+    [asset?.explorerTxLink, txId],
+  )
+
+  const handleSignTx = useCallback(async () => {
+    if (!isActionable) return
+
+    setIsSubmitting(true)
+
+    await onSignAndBroadcast!()
```

Comment:
this needs to be wrapped in a `Required<..>` in the props type definition rather than asserting here

Reply:
if it truly is optional you can do this to prevent a crash

```suggestion
    await onSignAndBroadcast?.()
```

Reply:
Yeah, we do have a discriminated union, so the non-null assertion is actually correct but TS is being derp:

https://github.com/shapeshift/web/blob/c8b8f4e9f0ca33607fc2ee4ad76cac9b894670d2/src/pages/RFOX/components/Shared/TransactionRow.tsx#L35
https://github.com/shapeshift/web/blob/c8b8f4e9f0ca33607fc2ee4ad76cac9b894670d2/src/pages/RFOX/components/Shared/SharedMultiStepStatus.tsx#L34-L37